### PR TITLE
Add Catalog

### PR DESCRIPTION
**Pull request by kandre**
- Checkboxes are working
- Tooltip are working

To do: Layer tree

### CODE REVIEW

File: js/catalog.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  $.displayLayer = function(layerBodId, visible) {
+    layers[layerBodId].setVisible(visible);
+  };
+
+  $.getLegend = function(layerBodId) {
+    var modalBody = $('#legend-modal').find('.modal-body');
+    modalBody.empty();
+    // Use JSONP for IE9
+    $.ajax({
+      url: location.protocol + '//api3.geo.admin.ch/rest/services/api/MapServer/' + layerBodId + '/legend?lang=de',
```
Comment (oterral) [inline js/catalog.js]: to be consistent for the demo use `en` everywhere

File: index.html [hunk 2]
```diff
@@ -20,12 +20,40 @@
       </div>
     </div>
     <div id="map"></div>
-    <div id="catalog">Catalog</div>
+    <div id="catalog">
+
+      <div id="MainMenu">
```
Comment (oterral) [inline index.html]: Use camel case for id

File: js/catalog.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  };
+
+  var catalogConfig = [
+    {layerBodId: 'ch.bafu.bundesinventare-amphibien', label: 'Amphibien Ortsfeste Objekte'},
+    {layerBodId: 'ch.bafu.bundesinventare-amphibien_wanderobjekte', label: 'Amphibien Wanderobjekte'},
+    {layerBodId: 'ch.bafu.bundesinventare-auen', label: 'Auengebiete'},
+    {layerBodId: 'ch.bafu.bundesinventare-bln', label: 'BLN'},
+    {layerBodId: 'ch.bafu.bundesinventare-flachmoore', label: 'Flachmoore'},
+    {layerBodId: 'ch.bafu.bundesinventare-hochmoore', label: 'Hochmoore'}
+  ];
```
Comment (oterral) [inline js/catalog.js]: Can you use the official layers pls . see list in the spec

Comment (kandre) [inline js/catalog.js]: yes those one are just for trial

File: js/catalog.js (hunk 1 above)
Comment (kandre) [inline js/catalog.js]: good idea and `de` from index.html will come from PR #10

**Comment by kandre** [comment]
Rebased

**Comment by kandre** [comment]
For your information, I saw on the getCap that @AFoletti added 3 'layers' for trial on wms-test: 

ch.blw.emapis-entwaesserung
ch.blw.emapis-seilbahnen
ch.blw.emapis-wegebau

I will however add all the requested layers in the catalog, therefore dont worry if the console becomes fuzzy with 'Resource interpreted as Image but transferred with MIME type text/xml' for some layers.

**Comment by kandre** [comment]
To do:
- [] use more less syntax for the css
- [] code cleaning

**Comment by kandre** [comment]
The wms-test server is way to slow, therefore I used some layers on wms-geo.admin.ch

**Comment by oterral** [comment]
can you rebased pls

**Comment by oterral** [comment]
@kandre don't forget to remove the `.swp` files and add it in .gitignore

**Comment by kandre** [comment]
Rebased

**Comment by kandre** [comment]
Here is a permalink : (http://10.220.4.46/~ltkan/web-emapis/?massnahme=ch.are.alpenkonvention)[http://10.220.4.46/~ltkan/web-emapis/?massnahme=ch.are.alpenkonvention]

**Comment by oterral** [comment]
can I review @kandre ?

**Comment by kandre** [comment]
I will center the map on the wms layers extent, in another PR. At the moment, we can see wms layers objects by typing 'Sur' in the search

**Comment by kandre** [comment]
@oterral It is ready for review

**Comment by kandre** [comment]
Updated

File: index.html [hunk 4]
```diff
@@ -30,16 +30,43 @@
       </div>
     </div>
     <div id="map"></div>
-    <div id="catalog">Katalog</div>
+    <div id="catalog">
+
```
Comment (kandre) [inline index.html]: I will remove this space

File: index.html [hunk 5]
```diff
@@ -30,16 +30,43 @@
...
+      <div id="mainMenu">
+        <div class="list-group panel">
+          <a href="#demo3" class="list-group-item list-group-item-success list-group-menu" data-toggle="collapse" data-parent="#MainMenu">eMAPIS</a>
+          <div class="collapse in" id="demo3">
+            <a href="#SubMenu1" class="list-group-item border-submenu" data-toggle="collapse" data-parent="#SubMenu1">Strukturverbesserungen<i class="fa fa-caret-down"></i></a>
+            <div class="collapse list-group-submenu my-class1 in" id="SubMenu1">
```
Comment (oterral) [inline index.html]: all id with camelCase pls

File: index.html [hunk 6]
```diff
@@ -30,16 +30,43 @@
...
+            </div>
+            <a href="#SubMenu2" class="list-group-item border-submenu" data-toggle="collapse" data-parent="#SubMenu2">Hintergrund Daten<i class="fa fa-caret-down"></i></a>
```
Comment (oterral) [inline index.html]: We don't have font-awesome for now you can remove the `<i>`

File: index.html [hunk 7]
```diff
@@ -30,16 +30,43 @@
...
+        </div>
+      </div>
+
+    </div>
+    <div class="modal fade" id="legend-modal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true" hidden-print="true">
+      <div class="modal-dialog">
+        <div class="modal-content">
+          <div class="modal-header">
+            <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
+            <h4 class="i" id="myModalLabel">Infobox</h4>
```
Comment (oterral) [inline index.html]: Avoid variable or id with `myXXXX`

File: index.html [hunk 8]
```diff
@@ -30,16 +30,43 @@
...
+          </div>
+          <div class="modal-body"></div>
+        </div>
+      </div>
+    </div>
     <script src="lib/jquery.min.js"></script>
     <script src="lib/bootstrap.min.js"></script>
     <script src="lib/typeahead.bundle.js"></script>
-    <script src="http://api3.geo.admin.ch/loader.js?lang=de"></script>
+    <script src="http://api3.geo.admin.ch/loader.js?mode=debug"></script>
```
Comment (oterral) [inline index.html]: Don't forget to remove debug mode

File: js/catalog.js [hunk 9]
```diff
@@ -0,0 +1,116 @@
+var addCatalog = function(map) {
+
+  var layers ={};
+  $.displayLayer = function(layerBodId, visible) {
+    layers[layerBodId].setVisible(visible);
+  };
+
+  $.getLegend = function(layerBodId) {
+    var modalBody = $('#legend-modal').find('.modal-body');
```
Comment (oterral) [inline js/catalog.js]: id with camleCase

File: js/catalog.js [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+
+  var nano = function(template, data) {
+    return template.replace(/\{([\w\.]*)\}/g,
+      function (str, key) {
+        var keys = key.split("."), value = data[keys.shift()];
+            $.each(keys, function () {
+                value = value[this];
+            });
+              return (value === null || value === undefined) ? "" : value;
+      }
```
Comment (oterral) [inline js/catalog.js]: weird indentations

File: js/catalog.js [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+  }
+  if (objPermalink.status) {
+    tmp.push('status=' + objPermalink.status);
+  }
+  if (objPermalink.projekt) {
+    tmp.push('projekt=' + objPermalink.projekt);
+  }
+  if (objPermalink.lang) {
+    tmp.push('lang=' + objPermalink.lang);
+  }
```
Comment (oterral) [inline js/catalog.js]: `lang` is not used by the layers no need to add it

File: js/catalog.js [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+    //{layerBodId: 'ch.blw.emapis-bewaesserung', label: 'Bewässerung', layerMenu: '0', layerLegend: 'hidden'},
+    //{layerBodId: 'ch.blw.emapis-elektrizitaetsversorgung', label: 'Elektrizitätsversorgung', layerMenu: '0', layerLegend: 'hidden'},
+    {layerBodId: 'ch.blw.emapis-entwaesserung', label: 'Entwässerung', layerMenu: '0', layerLegend: 'hidden'},
+    //{layerBodId: 'ch.blw.emapis-hochbau', label: 'Hochbau', layerMenu: '0', layerLegend: 'hidden'},
+    //{layerBodId: 'ch.blw.emapis-milchleitung', label: 'Milchleitung', layerMenu: '0', layerLegend: 'hidden'},
+    //{layerBodId: 'ch.blw.emapis-oekologie', label: 'Ökologie', layerMenu: '0', layerLegend: 'hidden'},
+    //{layerBodId: 'ch.blw.emapis-projektschwerpunkt', label: 'Projektschwerpunkt', layerMenu: '0', layerLegend: 'hidden'},
+    {layerBodId: 'ch.blw.emapis-seilbahnen', label: 'Seilbahnen', layerMenu: '0', layerLegend: 'hidden'},
+    //{layerBodId: 'ch.blw.emapis-wasserversorgung', label: 'Wasserversorgung', layerMenu: '0', layerLegend: 'hidden'},
+    {layerBodId: 'ch.blw.emapis-wegebau', label: 'Wegebau', layerMenu: '0', layerLegend: 'hidden'},
```
Comment (oterral) [inline js/catalog.js]: May be set these layers visible by default

File: js/catalog.js [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+          catalogConfig[j].isInPermalink = 'checked';
+        }
+      }
+    }
+  }
+
+  for (var i=0; i < catalogConfig.length; i++) {
+    var item = catalogConfig[i];
+    myClass = ['.my-class1', '.my-class2'];
+    $(myClass[item.layerMenu]).append(nano(tpl, item));
```
Comment (oterral) [inline js/catalog.js]: As in previous comment don't use `myXXX' varibales name

File: js/catalog.js [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+    if (item.layerMenu == 0) {
```
Comment (oterral) [inline js/catalog.js]: I think `layerMenu` is not very understandable maybe `layerType` is more appropriate

File: js/catalog.js [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+        })
+      });
+    } else if (item.layerMenu == 1) {
+      layers[item.layerBodId] = ga.layer.create(item.layerBodId);
+    }
+    if (item.isInPermalink == 'checked') {
+      layers[item.layerBodId].setVisible(true);
+    } else {
+      layers[item.layerBodId].setVisible(false);
+    };
```
Comment (oterral) [inline js/catalog.js]: One line of code to rule them all : `layers[item.layerBodId].setVisible((item.isInPermalink == 'checked'));`

File: less/catalog.less [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+      position: relative!important;
+      float: left;
+    }
+  }
+
+  .btn {
+    position: absolute;
+    right: 0px;
+    bottom: 2px;
+}
```
Comment (oterral) [inline less/catalog.less]: indentation

**Comment by oterral** [comment]
Minor comments otherwise lgtm , maybe you can add some comments in the code.

**Comment by kandre** [comment]
TYVM

**Comment by kandre** [comment]
Corrected

**Comment by kandre** [comment]
Some parts could be more functionnal and less linear but I can fix it later

File: index.html [hunk 17]
```diff
@@ -30,7 +30,31 @@
       </div>
     </div>
     <div id="map"></div>
-    <div id="catalog">Katalog</div>
+    <div id="catalog">
+      <div id="mainMenu">
+        <div class="list-group panel">
+          <a href="#demo3" class="list-group-item list-group-item-success list-group-menu" data-toggle="collapse" data-parent="#MainMenu">eMAPIS</a>
```
Comment (oterral) [inline index.html]: `mainMenu`

File: index.html [hunk 18]
```diff
@@ -30,7 +30,31 @@
...
+          <div class="collapse in" id="demo3">
+            <a href="#SubMenu1" class="list-group-item border-submenu" data-toggle="collapse" data-parent="#SubMenu1">Strukturverbesserungen</a>
```
Comment (oterral) [inline index.html]: `subMenu1`

File: index.html [hunk 19]
```diff
@@ -30,7 +30,31 @@
...
+            <div class="collapse list-group-submenu category1 in" id="subMenu1">
+            </div>
+            <a href="#SubMenu2" class="list-group-item border-submenu" data-toggle="collapse" data-parent="#SubMenu2">Hintergrund Daten</a>
```
Comment (oterral) [inline index.html]: `subMenu2`

**Comment by kandre** [comment]
New commit: Uncheck default layers if a layer in the permalink is fed

I had to fix a bug in the permalink.js: if no massnahme is given, the property res.massnahme is a void string. It is more consistent to have undefined instead.

File: index.html [hunk 20]
```diff
@@ -30,7 +30,31 @@
...
+          </div>
+        </div>
+      </div>
+    </div>
+    <div class="modal fade" id="legendModal" tabindex="-1" role="dialog" aria-labelledby="modalLabel" aria-hidden="true" hidden-print="true">
+      <div class="modal-dialog">
+        <div class="modal-content">
+          <div class="modal-header">
+            <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
+            <h4 class="i" id="modalLabel">Infobox</h4>
```
Comment (oterral) [inline index.html]: Is this class exist?

**Comment by kandre** [comment]
Rebased

**Comment by oterral** [comment]
All good thx

**Pull request merged**